apps/channels/tasks.py

```diff
@@ -146,3 +156,40 @@ def handle_api_message(
         user=user,
     )
     return channel.new_user_message(message)
+
+
+def handle_connect_messaging_message(payload: NewMessagePayload):
+    connect_channel_id = payload.get("channel_id")
+
+    try:
+        participant_data = ParticipantData.objects.prefetch_related("participant").get(
+            content_type=ContentType.objects.get_for_model(Experiment), system_metadata__channel_id=connect_channel_id
+        )
+
+        experiment_channel = ExperimentChannel.objects.prefetch_related("experiment").get(
+            platform=ChannelPlatform.CONNECT_MESSAGING, experiment__id=participant_data.object_id
+        )
+    except ParticipantData.DoesNotExist:
+        log.error(f"No participant data found for channel_id: {connect_channel_id}")
+        return
+    except ExperimentChannel.DoesNotExist:
+        log.error(f"No experiment channel found for participant channel_id: {connect_channel_id}")
+        return
```

Comment:
You could move this to the api view and pass in the `participant_data.id` and `experiment_channel.id` so that more validation happens in the view which is synchronous.